packages/labs/motion/src/spring.ts

```diff
@@ -0,0 +1,175 @@
+  }
+
+  /**
+   * The spring's current velocity in units / ms.
+   */
+  get currentVelocity(): number {
+    return this._spring.currentVelocity;
+  }
+
+  /**
+   * If the spring has reached its `toValue`, or if its velocity is below the
+   * `restVelocityThreshold`, it is considered at rest. If `stop()` is called
+   * during a simulation, both `isAnimating` and `isAtRest` will be false.
+   */
+  get isAtRest(): boolean {
+    return this._spring.isAtRest;
+  }
+
+  /**
+   * Whether or not the spring is currently emitting values.
+   *
+   * Note: this is distinct from whether or not it is at rest.
+   * See also `isAtRest`.
+   */
+  get isAnimating(): boolean {
+    return this._spring.isAnimating;
+  }
+
+  get toValue() {
+    return this._toValue;
+  }
+
+  set toValue(toValue: number) {
+    this._toValue = toValue;
+    this._spring.updateConfig({toValue});
+    if ((this._host as HTMLElement).isConnected) {
+      this._spring.start();
+    }
+  }
+
+  hostConnected() {
+    this._spring.start();
+  }
+
+  hostDisconnected() {
+    this._spring.stop();
+  }
+}
+
+export interface Position2D {
+  x: number;
+  y: number;
+}
+
+export interface Spring2DConfig
+  extends Omit<SpringConfig, 'fromValue' | 'toValue'> {
+  toPosition?: Position2D;
+  fromPosition?: Position2D;
+}
+
+/**
+ * A reactive controller that implements a 2D spring physics simulation by
+ * combining two 1D spring controllers.
+ */
+export class SpringController2D implements ReactiveController {
+  private _xSpring: SpringController;
+  private _ySpring: SpringController;
+
+  _toPosition: Position2D;
+
+  constructor(
+    host: ReactiveControllerHost & HTMLElement,
+    options?: Spring2DConfig
+  ) {
+    this._toPosition = options?.toPosition ?? {x: 0, y: 0};
+    this._xSpring = new SpringController(host, {
+      ...options,
+      fromValue: options?.fromPosition?.x,
+      toValue: this._toPosition.x,
+    });
+    this._ySpring = new SpringController(host, {
+      ...options,
+      fromValue: options?.fromPosition?.y,
+      toValue: this._toPosition.y,
+    });
+  }
+
+  get currentPosition() {
+    return {x: this._xSpring.currentValue, y: this._ySpring.currentValue};
+  }
+
+  /**
+   * The spring's current velocity in units / ms.
+   */
+  get currentVelocity(): number {
+    const dx = this._xSpring.currentVelocity;
+    const dy = this._ySpring.currentVelocity;
+    return Math.sqrt(dx ** 2 + dy ** 2);
+  }
+
+  /**
+   * If the spring has reached its `toValue`, or if its velocity is below the
+   * `restVelocityThreshold`, it is considered at rest. If `stop()` is called
+   * during a simulation, both `isAnimating` and `isAtRest` will be false.
+   */
+  get isAtRest(): boolean {
+    return this._xSpring.isAtRest && this._ySpring.isAtRest;
+  }
+
+  /**
+   * Whether or not the spring is currently emitting values.
+   *
+   * Note: this is distinct from whether or not it is at rest.
+   * See also `isAtRest`.
```

Comment:
Nit: JsDoc reference to `isAtRest` for API docs link. Doesn't need to be acted upon, but I think `[[isAtRest]]` will add a reference.

Ref: https://typedoc.org/guides/doccomments/#symbol-references

Reply:
Replaced with `{@link}`